xds/internal/server/listener_wrapper.go

```diff
@@ -331,15 +331,19 @@ func (l *listenerWrapper) handleRDSUpdate(update rdsHandlerUpdate) {
 	}
 	if update.err != nil {
 		if xdsresource.ErrType(update.err) == xdsresource.ErrorTypeResourceNotFound {
-			l.switchMode(nil, connectivity.ServingModeNotServing, update.err)
+			l.mu.Lock()
+			l.filterChains = nil
+			l.switchModeLocked(connectivity.ServingModeNotServing, update.err)
+			l.mu.Unlock()
 		}
 		// For errors which are anything other than "resource-not-found", we
 		// continue to use the old configuration.
 		return
 	}
 	atomic.StorePointer(&l.rdsUpdates, unsafe.Pointer(&update.updates))
```

Comment:
We need to verify somewhere that we are still watching this RDS update.  Probably at the top?